### ISSUE TITLE
kamailio error

### ISSUE THREAD

**Issue by andycol** [sipcapture/homer-installer#18]
new error i loaded a brand new vm and installed with the script and now kamailio shows

Mar 28 12:39:55 homer01 homer[12923]: ERROR: <core> [db.c:318]: db_do_init2(): could not add connection to the pool
Mar 28 12:39:55 homer01 homer[12923]: ERROR: sipcapture [sipcapture.c:1142]: child_init(): unable to connect to database [mysql://homer_user:homer_password@127...mode param.
Mar 28 12:39:55 homer01 homer[12923]: ERROR: <core> [core/sr_module.c:923]: init_mod_child(): Error while initializing module sipcapture (//usr/lib/x86_64-linu...capture.so)
Mar 28 12:39:55 homer01 homer[12923]: ERROR: <core> [core/pt.c:330]: fork_process(): init_child failed for process 1, pid 12923, "udp receiver child=0 sock=0.0.0.0:9060"
Mar 28 12:39:55 homer01 homer[12923]: CRITICAL: <core> [main.c:1606]: main_loop(): Cannot fork
Mar 28 12:39:55 homer01 homer[12949]: ERROR: db_mysql [km_my_con.c:124]: db_mysql_new_connection(): driver error: Access denied for user 'homer_user'@'localhos...sword: YES)
Mar 28 12:39:55 homer01 homer[12949]: ERROR: <core> [db.c:318]: db_do_init2(): could not add connection to the pool
Mar 28 12:39:55 homer01 homer[12949]: ERROR: sipcapture [sipcapture.c:1142]: child_init(): unable to connect to database [mysql://homer_user:homer_password@127...mode param.
Mar 28 12:39:55 homer01 homer[12950]: CRITICAL: <core> [core/pass_fd.c:277]: receive_fd(): EOF on 7
Mar 28 12:39:55 homer01 homer[12950]: CRITICAL: <core> [core/pass_fd.c:277]: receive_fd(): EOF on 23

**Comment by 00Asgaroth00** [comment]
this is because the installer script does not properly search/replace the mysql username/password in the kamailio configuration file, at least it didnt do it for the CentOS 7 section of the script, what os you running it on?

See issue #17 for some of the script changes I needed to make to get it working for me.

**Comment by andycol** [comment]
i am running debian 8

**Comment by 00Asgaroth00** [comment]
I'm guessting the same issue applies for you too, you can try change these lines in the installer script from:

```
    327                 # Replace values in template
    328                 perl -p -i -e "s/\{\{ LISTEN_PORT \}\}/$LISTEN_PORT/" $PATH_KAMAILIO_CFG
    329                 perl -p -i -e "s/\{\{ DB_PASS \}\}/$DB_PASS/" $PATH_KAMAILIO_CFG
    330                 perl -p -i -e "s/\{\{ DB_HOST \}\}/$DB_HOST/" $PATH_KAMAILIO_CFG
    331                 perl -p -i -e "s/\{\{ DB_USER \}\}/$DB_USER/" $PATH_KAMAILIO_CFG
```
to:
```
    327                 # Replace values in template
    328                 perl -p -i -e "s|^(#!substdef \"!HOMER_LISTEN_PORT)!.*|\1!${LISTEN_PORT}!g\"|" $PATH_KAMAILIO_CFG
    329                 perl -p -i -e "s|^(#!substdef \"!HOMER_DB_USER)!.*|\1!${DB_USER}!g\"|" $PATH_KAMAILIO_CFG
    330                 perl -p -i -e "s|^(#!substdef \"!HOMER_DB_PASSWORD)!.*|\1!${DB_PASS}!g\"|" $PATH_KAMAILIO_CFG
```

Then run the installer again and it should replace them for you.

**Comment by andycol** [comment]
that worked!

you should add a pull request

**Comment by tandyuk** [comment]
Just had this exact issue on a fresh deb8 machine.

**Comment by 00Asgaroth00** [comment]
This issue is corrected in pull request #23
